### PR TITLE
Fix support for adding default namespaces when wikis are created

### PR DESCRIPTION
**Pull request by paladox**

### CODE REVIEW

**Comment by JohnFLewis** [comment]
Why use ManageWiki.php?

File: includes/ManageWiki.php [hunk 1]
```diff
@@ -256,4 +256,59 @@ public static function modifyPermissions( $group, $addp = [], $removep = [], $ad
...
+
+	public static function defaultCanonicalNamespaces() {
+		global $wgCreateWikiDatabase;
+
+		$dbr = wfGetDB( DB_REPLICA, [], $wgCreateWikiDatabase );
+
+		$res = $dbr->select(
+			'mw_namespaces',
+			'ns_namespace_id',
+			[ 'ns_dbname' => 'default' ]
```
Comment (JohnFLewis) [inline includes/ManageWiki.php]: Please expand this out for readability

File: includes/ManageWiki.php [hunk 2]
```diff
@@ -256,4 +256,59 @@ public static function modifyPermissions( $group, $addp = [], $removep = [], $ad
...
+			],
+			[
+				'ns_dbname' => 'default',
+				'ns_namespace_id' => $namespace
+			]
+		);
+		
+		$ns = [];
+
+		$ns['ns_namespace_name'] = json_decode( $res->ns_namespace_name, true );
```
Comment (JohnFLewis) [inline includes/ManageWiki.php]: Why are you json_decoding non-json encoded things?

File: includes/ManageWiki.php [hunk 3]
```diff
@@ -256,4 +256,59 @@ public static function modifyPermissions( $group, $addp = [], $removep = [], $ad
...
+		$ns['ns_searchable'] = json_decode( $res->ns_searchable, true );
+		$ns['ns_subpages'] = json_decode( $res->ns_subpages, true );
+		$ns['ns_content'] = json_decode( $res->ns_content, true );
+		$ns['ns_protection'] = json_decode( $res->ns_protection, true );
+		$ns['ns_aliases'] = json_decode( $res->ns_aliases, true );
```
Comment (JohnFLewis) [inline includes/ManageWiki.php]: Why decode? Are you manipulating this?

File: includes/ManageWikiHooks.php [hunk 4]
```diff
@@ -154,39 +154,39 @@ public static function onCreateWikiCreation( $dbname, $private ) {
...
 			$dbw = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
-			$namespaces = $wgCanonicalNamespaceNames + [ 0 => '<Main>' ];
 
-			foreach ( $namespaces as $id => $name ) {
-				if ( $id < 0 ) {
-					// Don't include imaginary
-					continue;
-				}
+			foreach ( $defaultCanonicalNamespaces as $newnamespace ) {
+				$namespacesarray = ManageWiki::defaultNamespaces( $newnamespace );
```
Comment (JohnFLewis) [inline includes/ManageWikiHooks.php]: camelCase

File: includes/ManageWikiHooks.php [hunk 5]
```diff
@@ -154,39 +154,39 @@ public static function onCreateWikiCreation( $dbname, $private ) {
...
 
-				$matchedNSKeys = array_keys( $wgNamespaceAliases, $id );
 				$nsAliases = [];
-
-				foreach ( $matchedNSKeys as $o => $n ) {
-					$nsAliases[] = $n;
+				if ( is_array( $namespacesarray['ns_aliases'] ) ) {
```
Comment (JohnFLewis) [inline includes/ManageWikiHooks.php]: This whole thing is unnecessary if you don’t decide the json

File: includes/ManageWiki.php [hunk 6]
```diff
@@ -256,4 +256,59 @@ public static function modifyPermissions( $group, $addp = [], $removep = [], $ad
...
+
+		return $namespaces;
+	}
+
+	public static function defaultNamespaces( $namespace ) {
+		global $wgCreateWikiDatabase;
+
+		$dbr = wfGetDB( DB_REPLICA, [], $wgCreateWikiDatabase );
+
+		$res = $dbr->selectRow(
```
Comment (JohnFLewis) [inline includes/ManageWiki.php]: Also this isn’t a result, it’s a row.

File: includes/ManageWiki.php (hunk 2 above)
Comment (paladox) [reply includes/ManageWiki.php]: Was following how it's done for groups, but don't you need to decode to foreach over the value?

File: includes/ManageWiki.php (hunk 1 above)
Comment (paladox) [reply includes/ManageWiki.php]: Thanks, done.

File: includes/ManageWiki.php (hunk 2 above)
Comment (paladox) [reply includes/ManageWiki.php]: Thanks, done.

File: includes/ManageWikiHooks.php (hunk 4 above)
Comment (paladox) [reply includes/ManageWikiHooks.php]: Thanks, done.

File: includes/ManageWikiHooks.php (hunk 5 above)
Comment (paladox) [reply includes/ManageWikiHooks.php]: Thanks, done.

File: includes/ManageWiki.php (hunk 6 above)
Comment (paladox) [reply includes/ManageWiki.php]: Thanks, done.

File: includes/ManageWiki.php (hunk 3 above)
Comment (paladox) [reply includes/ManageWiki.php]: I've removed the json decoding, thanks, done.

**Comment by paladox** [comment]
@JohnFLewis was following how it was done for groups.

**Comment by JohnFLewis** [comment]
Except that this is still isn’t in ManageWikiNamespaces.php it looks fine.

Though I strong recommend against passing $row, it seems in friendly to pass an object esp when 100% of its uses will be as an array

**Comment by paladox** [comment]
@JohnFLewis so what i put in ManageWiki i should put in ManageWikiNamespaces.php?

**Comment by paladox** [comment]
also "Though I strong recommend against passing $row, it seems un friendly to pass an object esp when 100% of its uses will be as an array" What do you mean? What other way to do this?

**Comment by JohnFLewis** [comment]
Yes an an array?

**Pull request merged**